# Project 4

In this project, you will summarize and present your analysis from Projects 1-3.

### Intro: Write a problem Statement/ Specific Aim for this project

Answer: Investigate the relationship(s) between gaining admittance into a graduate school (still not quite sure which one) based on an undergraduate students GRA, GRE, and the respective prestige of their undergraduate school.

### Dataset:  Write up a description of your data and any cleaning that was completed

In [8]:
import pandas as pd
df = pd.read_csv("../assets/admissions.csv")

In [9]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 400 entries, 0 to 399
Data columns (total 4 columns):
admit       400 non-null int64
gre         398 non-null float64
gpa         398 non-null float64
prestige    399 non-null float64
dtypes: float64(3), int64(1)
memory usage: 12.6 KB


In [10]:
df = df.dropna()

In [12]:
df.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 397 entries, 0 to 399
Data columns (total 4 columns):
admit       397 non-null int64
gre         397 non-null float64
gpa         397 non-null float64
prestige    397 non-null float64
dtypes: float64(3), int64(1)
memory usage: 15.5 KB


In [13]:
df.describe()

admit         gre         gpa    prestige
count  397.000000  397.000000  397.000000  397.000000
mean     0.317380  587.858942    3.392242    2.488665
std      0.466044  115.717787    0.380208    0.947083
min      0.000000  220.000000    2.260000    1.000000
25%      0.000000  520.000000    3.130000    2.000000
50%      0.000000  580.000000    3.400000    2.000000
75%      1.000000  660.000000    3.670000    3.000000
max      1.000000  800.000000    4.000000    4.000000

Answer: Dataset consists of 400 entries. Some observations were missing, these (3 entries) were simply dropped.

admittance = ranged from 0-1 (binary)

gre = ranged from 220 - 800 (integer)

gpa = ranged from 2.26 - 4.0 (float)

prestige = ranged from 1-4 (integer)

### Demo: Provide a table that explains the data by admission status

Mean (STD) or counts by admission status for each variable 

| Not Admitted | Admitted
---| ---|---
GPA | 3.2(0.3)  | 3.8(0.3)
GRE |500(100) | 700(100)
Prestige 1 | 50| 100
Prestige 2 | 100 | 50
Prestige 3 |125 | 25
Prestige 4 |200 | 10

### Methods: Write up the methods used in your analysis

Answer:

Initially attempted to use sklearn's LogisticRegression model, which I eventually gave up on based on since I couldnt' really read the results (need to read up on this more...)

Switched to the logistical regression model from the statsmodel. Calculated admittance probability based on the undergraduate school's prestige (controlling for GRE/GPA).

### Results: Write up your results

Answer:

1. Students from a prestige1 undergraduate school have 3.07949... times a greater chance of being admitted into a graduate program than students from a non-prestige1 school.

2. The odds of gaining admittance for a student from a prestige4 undergraduate school are 58% less than those from a non-prestige4 undergraduate school. :(

3. Students from a prestige2 undergrad school have odds of ~0.51 of gaining admittance compared to students from prestige1 undergrad schools.

### Visuals: Provide a table or visualization of these results

In [27]:
## not sure how to actually generate the below, but I found them sitting in the assets folder. -mt

<img src='../assets/images/figure1.jpg' height= 50% width= 50%>

<img src='../assets/images/figure2.jpg' height= 50% width= 50%>

### Discussion: Write up your discussion and future steps

Answer: 

In summation, there was a definite skew/bias towards this graduate school accepting more students from fancier/more prestigious undergraduate schools.

Future steps would largely depend on the ability to get more data from this or a similar dataset. Would be interesting to see if a student's ethnicity (whether explicit or implied by their name) plays into admission bias.